activity_calendar/feeds.py

```diff
@@ -0,0 +1,237 @@
+    ("tz_id",           'tzid')
+]
+
+ICAL_DAYLIGHT_FIELDS_MAP = [
+    ("tzoffset_from", 'tzoffsetfrom'),
+    ("tzoffset_to", 'tzoffsetto'),
+    ("tzname", 'tzname'),
+    ("dtstart", 'dtstart'),
+    ("rrule", 'rrule'),
+]
+
+ICAL_VTIMEZONE_ITEMS_MAP = [
+    ("daylight", TimezoneDaylight),
+    ("standard", TimezoneStandard),
+]
+
+class ExtendedICal20Feed(ICal20Feed):
+    """
+    iCalendar 2.0 Feed implementation that also supports VTIMEZONE.
+    """
+
+    def write_items(self, calendar):
+        """
+        Writes the feed to the specified file in the
+        specified encoding.
+        """
+        tz_info = Timezone()
+        for ifield, efield in ICAL_VTIMEZONE_FIELDS_MAP:
+            val = self.feed.get(ifield)
+            if val is not None:
+                tz_info.add(efield, val)
+        
+        for ifield, tz_class in ICAL_VTIMEZONE_ITEMS_MAP:
+            is_present = self.feed.get(ifield)
+            if val:
+                tz_dst = tz_class()
+                for idaylightfield, efield in ICAL_DAYLIGHT_FIELDS_MAP:
+                    val = self.feed.get(ifield + "_" + idaylightfield)
+                    if val is not None:
+                        tz_dst.add(efield, val)
+                tz_info.add_component(tz_dst)
+
+        calendar.add_component(tz_info)
+
+        super().write_items(calendar)
+
+
+class CESTEventFeed(ICalFeed):
+    """
+    A simple event calender
+    Please refer the docs for the full list of options:
+    https://django-ical.readthedocs.io/en/latest/usage.html#property-reference-and-extensions
+    """
+    feed_type = ExtendedICal20Feed
+
+    product_id = '-//Squire//Activity Calendar//EN'
+    file_name = "knights-calendar.ics"
+
+    def title(self):
+        # TODO: unhardcode
+        return "Activiteiten Agenda - Knights"
+
+    def description(self):
+        # TODO: unhardcode
+        return "Knights of the Kitchen Table Activiteiten en Evenementen."
+
+    def method(self):
+        return "PUBLISH"
+
+    def timezone(self):
+        return settings.TIME_ZONE
+    
+    #######################################################
+    # Timezone information (Daylight-saving time, etc.)
+    def vtimezone_x_lic_location(self):
+        return settings.TIME_ZONE
+    
+    vtimezone_tz_id = settings.TIME_ZONE
+
+    # Summer Time
+    vtimezone_daylight_tzoffset_from = timedelta(hours=1)
+    vtimezone_daylight_tzoffset_to = timedelta(hours=2)
+    vtimezone_daylight_tzname = "CEST"
+    vtimezone_daylight_dtstart = datetime(1970, 3, 29, 20)
+    vtimezone_daylight_rrule = build_rrule_from_text("FREQ=YEARLY;BYMONTH=3;BYDAY=-1SU")
+
+    # Winter Time
+    vtimezone_standard_tzoffset_from = timedelta(hours=2)
+    vtimezone_standard_tzoffset_to = timedelta(hours=1)
+    vtimezone_standard_tzname = "CET"
+    vtimezone_standard_dtstart = datetime(1970, 10, 25, 3)
+    vtimezone_standard_rrule = build_rrule_from_text("FREQ=YEARLY;BYMONTH=10;BYDAY=-1SU")
```

Comment:
Isn't there a nicer way to do this? Timezones are annoying, can't you refer to the Olsen Timezone Database to retrieve this data? Otherwise we'll need to patch this app when DST introduction is moved or abolished